info/diktat-kotlin-coding-style-guide-en.md

```diff
@@ -1641,3 +1642,249 @@ Good example:
+So in your code you should try to use single `init` block to reduce the complexity. In case you need to do some logging or make some calculations before the assignment 
+of some class property - you can use powerful functional programming. This will reduce the possibility of the error, when occasioanlly someone will change the order of your `init` blocks. 
+And it will make the logic of the code more coupled. It is always enough to use one `init` block to implement your idea in Kotlin.
+
+Bad example:
+```kotlin
+class YourClass(var name: String) {    
+    init {
+        println("First initializer block that prints ${name}")
+    }
+    
+    val property = "Property: ${name.length}".also(::println)
+    
+    init {
+        println("Second initializer block that prints ${name.length}")
+    }
+}
+```
+
+Good example:
+```kotlin
+class YourClass(var name: String) {
+    init {
+        println("First initializer block that prints ${name}")
+    }
+
+    val property = "Property: ${name.length}".also { prop ->
+        println(prop)
+        println("Second initializer block that prints ${name.length}")
+    }
+}
+```
+
+### <a name="r6.1.5"></a> Rule 6.1.5: Explicit supertype qualification should not be used if there is not clash between called methods
+Bad example:
+```kotlin
+open class Rectangle {
+    open fun draw() { /* ... */ }
+}
+
+class Square() : Rectangle() {
+    override fun draw() {
+        super<Rectangle>.draw() // no need in super<Rectangle> here
+    }
+}
+```
+
+### <a name="r6.1.6"></a> Rule 6.1.6: Abstract class should have at least one abstract method
+Abstract classes are used to force a developer to implement some of its parts in its inheritors.
+In case when abstract class has no abstract methods - then it was set `abstract` incorrectly and can be converted to a normal class.
+Bad example:
+```kotlin
+abstract class NotAbstract {
+    fun foo() {}
+    
+    fun test() {}
+}
+```
+
+Good example:
+```kotlin
+abstract class NotAbstract {
+    abstract fun foo()
+    
+    fun test() {}
+}
+
+// OR
+class NotAbstract {
+    fun foo() {}
+    
+    fun test() {}
+}
+```
+
+
+### <a name="r6.1.7"></a> Rule 6.1.7: in case of using "implicit backing property" scheme, the name of real and back property should be the same
+Kotlin has a mechanism of [backing properties](https://kotlinlang.org/docs/reference/properties.html#backing-properties).
+In some cases implicit backing is not enough and it should be done explicitly:
+```kotlin
+private var _table: Map<String, Int>? = null
+val table: Map<String, Int>
+    get() {
+        if (_table == null) {
+            _table = HashMap() // Type parameters are inferred
+        }
+        return _table ?: throw AssertionError("Set to null by another thread")
+    }
+```
+
+In this case the name of backing property (`_table`) should be the same to the name of real property (`table`), but should have underscore (`_`) prefix.
```

Comment:
Will this clash with requirements of identifier naming chapter?

Reply:
yeah, hm, that is true - I will mention it